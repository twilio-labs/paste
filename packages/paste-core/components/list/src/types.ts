import type {TextProps} from '@twilio-paste/text';
import type {Space} from '@twilio-paste/style-props';

export type AsTags = 'ol' | 'ul';

export interface BaseListProps extends React.OlHTMLAttributes<HTMLElement>, Pick<TextProps, 'element'> {
  className?: never;
  style?: never;
  as: AsTags;
  marginTop?: Space;
  marginBottom?: Space;
}

export interface ListItemProps extends React.LiHTMLAttributes<HTMLLIElement>, Pick<TextProps, 'element'> {
  className?: never;
  style?: never;
}

export type OrderedListProps = Omit<BaseListProps, 'as'>;

export type UnorderedListProps = Omit<BaseListProps, 'as'>;
