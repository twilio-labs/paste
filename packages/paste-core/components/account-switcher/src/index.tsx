export * from "./AccountSwitcher";
export * from "./AccountSwitcherGroup";
export * from "./AccountSwitcherSeparator";
export * from "./AccountSwitcherItem";
export * from "./AccountSwitcherItemRadio";
export * from "./AccountSwitcherBadge";
export * from "./useAccountSwitcher";
