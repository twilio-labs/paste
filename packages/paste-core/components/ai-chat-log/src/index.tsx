export { AIChatMessage } from "./AIChatMessage";
export type { AIChatMessageProps } from "./AIChatMessage";
export { AIChatMessageActionGroup } from "./AIChatMessageActionGroup";
export type { AIChatMessageActionGroupProps } from "./AIChatMessageActionGroup";
export { AIChatMessageBody } from "./AIChatMessageBody";
export type { AIChatMessageBodyProps } from "./AIChatMessageBody";
export { AIChatMessageAuthor } from "./AIChatMessageAuthor";
export type { AIChatMessageAuthorProps } from "./AIChatMessageAuthor";
export { AIChatMessageActionCard } from "./AIChatMessageActionCard";
export type { AIChatMessageActionCardProps } from "./AIChatMessageActionCard";
export { AIChatMessageLoading } from "./AIChatMessageLoading";
export type { AIChatMessageLoadingProps } from "./AIChatMessageLoading";

export { AIChatLog } from "./AIChatLog";
export type { AIChatLogProps } from "./AIChatLog";
export { useAIChatLogger } from "./useAIChatLogger";
export type { UseAIChatLogger, AIPartialIDChat, AIChat } from "./useAIChatLogger";
export { AIChatLogger } from "./AIChatLogger";
export type { AIChatLoggerProps } from "./AIChatLogger";

export type { AIMessageVariants } from "./AIMessageContext";
