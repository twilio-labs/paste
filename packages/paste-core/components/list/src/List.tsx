import * as React from 'react';
import * as PropTypes from 'prop-types';
import {Text} from '@twilio-paste/text';
import {isMarginTokenProp} from '@twilio-paste/style-props';
import type {AsTags, BaseListProps} from './types';

const List = React.forwardRef<HTMLOListElement | HTMLUListElement, BaseListProps>(
  ({as, children, element = 'LIST', ...props}, ref) => {
    return (
      <Text
        {...props}
        as={as}
        color="colorText"
        element={element}
        fontSize="fontSize30"
        fontWeight="fontWeightNormal"
        lineHeight="lineHeight40"
        marginLeft="space70"
        ref={ref}
      >
        {children}
      </Text>
    );
  }
);

List.displayName = 'List';

List.propTypes = {
  as: PropTypes.oneOf(['ol', 'ul'] as AsTags[]).isRequired,
  element: PropTypes.string,
  marginTop: isMarginTokenProp,
  marginBottom: isMarginTokenProp,
};

export {List};
