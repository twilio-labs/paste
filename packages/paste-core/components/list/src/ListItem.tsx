import * as React from 'react';
import * as PropTypes from 'prop-types';
import {Text, safelySpreadTextProps} from '@twilio-paste/text';
import type {ListItemProps} from './types';

const ListItem = React.forwardRef<HTMLLIElement, ListItemProps>(({children, element = 'LIST_ITEM', ...props}, ref) => {
  return (
    <Text
      {...safelySpreadTextProps(props)}
      as="li"
      color="colorText"
      element={element}
      fontSize="fontSize30"
      fontWeight="fontWeightNormal"
      lineHeight="lineHeight40"
      marginBottom="space30"
      marginTop="space30"
      ref={ref}
    >
      {children}
    </Text>
  );
});

ListItem.displayName = 'ListItem';

ListItem.propTypes = {
  element: PropTypes.string,
};

export {ListItem};
