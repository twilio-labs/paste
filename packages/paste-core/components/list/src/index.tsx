import * as React from 'react';
import {Text, safelySpreadTextProps} from '@twilio-paste/text';
import {Space} from '@twilio-paste/style-props';

interface BaseListProps extends React.OlHTMLAttributes<''> {
  className?: never;
  style?: never;
  as: 'ol' | 'ul';
  marginTop?: Space;
  marginBottom?: Space;
}

const List: React.FC<BaseListProps> = ({as, children, ...props}) => {
  return (
    <Text
      {...props}
      as={as}
      marginLeft="space70"
      fontSize="fontSize30"
      lineHeight="lineHeight30"
      fontWeight="fontWeightNormal"
      textColor="colorText"
    >
      {children}
    </Text>
  );
};
List.displayName = 'List';

export type OrderedListProps = Omit<BaseListProps, 'as'>;
const OrderedList: React.FC<OrderedListProps> = ({children, marginTop, marginBottom, ...props}) => {
  return (
    <List {...safelySpreadTextProps(props)} as="ol" marginTop={marginTop} marginBottom={marginBottom}>
      {children}
    </List>
  );
};
OrderedList.defaultProps = {
  marginBottom: 'space70',
};
OrderedList.displayName = 'OrderedList';

export type UnorderedListProps = Omit<BaseListProps, 'as'>;
const UnorderedList: React.FC<UnorderedListProps> = ({children, marginTop, marginBottom, ...props}) => {
  return (
    <List {...safelySpreadTextProps(props)} as="ul" marginTop={marginTop} marginBottom={marginBottom}>
      {children}
    </List>
  );
};
UnorderedList.defaultProps = {
  marginBottom: 'space70',
};
UnorderedList.displayName = 'UnorderedList';

export interface ListItemProps extends React.LiHTMLAttributes<''> {
  className?: never;
  style?: never;
}

const ListItem: React.FC<ListItemProps> = props => {
  return (
    <Text
      {...safelySpreadTextProps(props)}
      as="li"
      marginTop="space30"
      marginBottom="space30"
      fontSize="fontSize30"
      lineHeight="lineHeight30"
      fontWeight="fontWeightNormal"
      textColor="colorText"
    >
      {props.children}
    </Text>
  );
};
ListItem.displayName = 'ListItem';

export {OrderedList, UnorderedList, ListItem};
