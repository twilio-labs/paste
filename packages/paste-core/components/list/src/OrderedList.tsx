import * as React from 'react';
import * as PropTypes from 'prop-types';
import {safelySpreadTextProps} from '@twilio-paste/text';
import {isMarginTokenProp} from '@twilio-paste/style-props';
import {List} from './List';
import type {OrderedListProps} from './types';

const OrderedList = React.forwardRef<HTMLOListElement, OrderedListProps>(
  ({children, element = 'ORDERED_LIST', marginTop, marginBottom = 'space70', ...props}, ref) => {
    return (
      <List
        {...safelySpreadTextProps(props)}
        as="ol"
        element={element}
        marginTop={marginTop}
        marginBottom={marginBottom}
        ref={ref}
      >
        {children}
      </List>
    );
  }
);

OrderedList.displayName = 'OrderedList';

OrderedList.propTypes = {
  element: PropTypes.string,
  marginTop: isMarginTokenProp,
  marginBottom: isMarginTokenProp,
};

export {OrderedList};
