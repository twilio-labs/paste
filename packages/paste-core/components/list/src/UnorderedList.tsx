import * as React from 'react';
import * as PropTypes from 'prop-types';
import {safelySpreadTextProps} from '@twilio-paste/text';
import {isMarginTokenProp} from '@twilio-paste/style-props';
import {List} from './List';
import type {UnorderedListProps} from './types';

const UnorderedList = React.forwardRef<HTMLUListElement, UnorderedListProps>(
  ({children, element = 'UNORDERED_LIST', marginTop, marginBottom = 'space70', ...props}, ref) => {
    return (
      <List
        {...safelySpreadTextProps(props)}
        as="ul"
        element={element}
        marginTop={marginTop}
        marginBottom={marginBottom}
        ref={ref}
      >
        {children}
      </List>
    );
  }
);

UnorderedList.displayName = 'UnorderedList';

UnorderedList.propTypes = {
  element: PropTypes.string,
  marginTop: isMarginTokenProp,
  marginBottom: isMarginTokenProp,
};

export {UnorderedList};
