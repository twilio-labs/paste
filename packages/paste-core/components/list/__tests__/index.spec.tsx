import * as React from 'react';
import {render} from 'react-dom';
import renderer from 'react-test-renderer';
import {ReactWrapper, mount} from 'enzyme';
import {axe} from 'jest-axe';
import {Theme} from '@twilio-paste/theme';
import {OrderedList, UnorderedList, ListItem} from '../src';

describe('Ordered List', () => {
  it('should render a plain ordered list wrapper', () => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <OrderedList>Children</OrderedList>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have no accessibility violations', async () => {
    const container = document.createElement('div');
    document.body.append(container);
    render(
      <Theme.Provider theme="console">
        <OrderedList>
          <ListItem>item</ListItem>
        </OrderedList>
      </Theme.Provider>,
      container
    );
    const results = await axe(document.body);
    expect(results).toHaveNoViolations();
  });

  it('should filter out styling props', () => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <OrderedList margin="space40">Children</OrderedList>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should allow marginTop and marginBottom styling props', () => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <OrderedList marginTop="space40" marginBottom="space40">
            Children
          </OrderedList>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should allow aria attributes', () => {
    const wrapper: ReactWrapper = mount(<OrderedList aria-expanded="true">Children</OrderedList>);
    expect(wrapper.exists('ol[aria-expanded="true"]')).toEqual(true);
  });

  it('should allow data attributes', () => {
    const wrapper: ReactWrapper = mount(<OrderedList data-test="test-hook">Children</OrderedList>);
    expect(wrapper.exists('ol[data-test="test-hook"]')).toEqual(true);
  });

  it('should allow HTML attributes', () => {
    const wrapper: ReactWrapper = mount(<OrderedList title="random title">Children</OrderedList>);
    expect(wrapper.exists('ol[title="random title"]')).toEqual(true);
  });
});

describe('Unordered List', () => {
  it('should render a plain unordered list wrapper', () => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <UnorderedList>Children</UnorderedList>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have no accessibility violations', async () => {
    const container = document.createElement('div');
    document.body.append(container);
    render(
      <Theme.Provider theme="console">
        <UnorderedList>
          <ListItem>item</ListItem>
        </UnorderedList>
      </Theme.Provider>,
      container
    );
    const results = await axe(document.body);
    expect(results).toHaveNoViolations();
  });

  it('should filter out styling props', () => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <UnorderedList textColor="textColorSuccess">Children</UnorderedList>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should allow aria attributes', () => {
    const wrapper: ReactWrapper = mount(<UnorderedList aria-label="label">Children</UnorderedList>);
    expect(wrapper.exists('ul[aria-label="label"]')).toEqual(true);
  });

  it('should allow data attributes', () => {
    const wrapper: ReactWrapper = mount(<UnorderedList data-rando="test-hook">Children</UnorderedList>);
    expect(wrapper.exists('ul[data-rando="test-hook"]')).toEqual(true);
  });

  it('should allow HTML attributes', () => {
    const wrapper: ReactWrapper = mount(<UnorderedList title="random title">Children</UnorderedList>);
    expect(wrapper.exists('ul[title="random title"]')).toEqual(true);
  });
});

describe('ListItem', () => {
  it('should render a plain list item', () => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <ListItem>Children</ListItem>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should filter out styling props', () => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <ListItem padding="textColorSuccess">Children</ListItem>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should allow aria attributes', () => {
    const wrapper: ReactWrapper = mount(<ListItem aria-pressed="true">Children</ListItem>);
    expect(wrapper.exists('li[aria-pressed="true"]')).toEqual(true);
  });

  it('should allow data attributes', () => {
    const wrapper: ReactWrapper = mount(<ListItem data-rando="test-hook">Children</ListItem>);
    expect(wrapper.exists('li[data-rando="test-hook"]')).toEqual(true);
  });

  it('should allow HTML attributes', () => {
    const wrapper: ReactWrapper = mount(<ListItem title="random title">Children</ListItem>);
    expect(wrapper.exists('li[title="random title"]')).toEqual(true);
  });
});
