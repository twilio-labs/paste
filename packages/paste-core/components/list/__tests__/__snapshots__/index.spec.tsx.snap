// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should filter out styling props 1`] = `
.emotion-1 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-1 *,
.emotion-1 *::after,
.emotion-1 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  margin: 0;
  padding: 0;
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
  font-size: 0.875rem;
  line-height: 1.5rem;
  font-weight: 400;
  color: #282a2b;
}

<div
  className="emotion-1"
>
  <li
    className="emotion-0"
    fontSize="fontSize30"
    fontWeight="fontWeightNormal"
  >
    Children
  </li>
</div>
`;

exports[`ListItem should render a plain list item 1`] = `
.emotion-1 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-1 *,
.emotion-1 *::after,
.emotion-1 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  margin: 0;
  padding: 0;
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
  font-size: 0.875rem;
  line-height: 1.5rem;
  font-weight: 400;
  color: #282a2b;
}

<div
  className="emotion-1"
>
  <li
    className="emotion-0"
    fontSize="fontSize30"
    fontWeight="fontWeightNormal"
  >
    Children
  </li>
</div>
`;

exports[`Ordered List should allow marginTop and marginBottom styling props 1`] = `
.emotion-1 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-1 *,
.emotion-1 *::after,
.emotion-1 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  margin: 0;
  padding: 0;
  margin-top: 0.75rem;
  margin-bottom: 0.75rem;
  margin-left: 1.5rem;
  font-size: 0.875rem;
  line-height: 1.5rem;
  font-weight: 400;
  color: #282a2b;
}

<div
  className="emotion-1"
>
  <ol
    className="emotion-0"
    fontSize="fontSize30"
    fontWeight="fontWeightNormal"
  >
    Children
  </ol>
</div>
`;

exports[`Ordered List should filter out styling props 1`] = `
.emotion-1 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-1 *,
.emotion-1 *::after,
.emotion-1 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  margin: 0;
  padding: 0;
  margin-bottom: 1.5rem;
  margin-left: 1.5rem;
  font-size: 0.875rem;
  line-height: 1.5rem;
  font-weight: 400;
  color: #282a2b;
}

<div
  className="emotion-1"
>
  <ol
    className="emotion-0"
    fontSize="fontSize30"
    fontWeight="fontWeightNormal"
  >
    Children
  </ol>
</div>
`;

exports[`Ordered List should render a plain ordered list wrapper 1`] = `
.emotion-1 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-1 *,
.emotion-1 *::after,
.emotion-1 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  margin: 0;
  padding: 0;
  margin-bottom: 1.5rem;
  margin-left: 1.5rem;
  font-size: 0.875rem;
  line-height: 1.5rem;
  font-weight: 400;
  color: #282a2b;
}

<div
  className="emotion-1"
>
  <ol
    className="emotion-0"
    fontSize="fontSize30"
    fontWeight="fontWeightNormal"
  >
    Children
  </ol>
</div>
`;

exports[`Unordered List should filter out styling props 1`] = `
.emotion-1 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-1 *,
.emotion-1 *::after,
.emotion-1 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  margin: 0;
  padding: 0;
  margin-bottom: 1.5rem;
  margin-left: 1.5rem;
  font-size: 0.875rem;
  line-height: 1.5rem;
  font-weight: 400;
  color: #282a2b;
}

<div
  className="emotion-1"
>
  <ul
    className="emotion-0"
    fontSize="fontSize30"
    fontWeight="fontWeightNormal"
  >
    Children
  </ul>
</div>
`;

exports[`Unordered List should render a plain unordered list wrapper 1`] = `
.emotion-1 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-1 *,
.emotion-1 *::after,
.emotion-1 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  margin: 0;
  padding: 0;
  margin-bottom: 1.5rem;
  margin-left: 1.5rem;
  font-size: 0.875rem;
  line-height: 1.5rem;
  font-weight: 400;
  color: #282a2b;
}

<div
  className="emotion-1"
>
  <ul
    className="emotion-0"
    fontSize="fontSize30"
    fontWeight="fontWeightNormal"
  >
    Children
  </ul>
</div>
`;
